src/bvals/bvals.hpp

```diff
@@ -151,6 +152,10 @@ class BoundaryValues : public BoundaryBase, // public BoundaryPhysics,
                                             int sk, int ek);
   void ProlongateGhostCells(const NeighborBlock &nb, int si, int ei, int sj, int ej,
                             int sk, int ek);
+  void ComputeRestrictionBounds_(const NeighborBlock &nb, int &nis, int &nie, int &njs,
+                                 int &nje, int &nks, int &nke);
+  void ComputeProlongationBounds_(const NeighborBlock &nb, int &si, int &ei, int &sj,
+                                  int &ej, int &sk, int &ek);
```

Comment:
I would recommend using IndexRange here instead of specifying every index bound individually, it will help prevent future users from putting them in, in the wrong order.

Reply:
This is a good idea.